### PR TITLE
Security update

### PR DESCRIPTION
**Pull request by GJRTimmer**
Renamed Security => VaadinSecurity to avoid naming conflict with Spring-security.

VaadinSecurity (Former Security) changed from implementation to interface.
Implementation of former Security => GenericVaadinSecurity

Split VaadinSecurity (former Security) implementation; moved generic to AbstractVaadinSecurity
Added interface VaadinSecurityContext implementation => AbstractVaadinSecurity
PR #117 implemented => AbstractVaadinSecurity with interface VaadinSecurityContext

Added VaadinSecurityAware
Added VaadinSecurityContextAware
Added bean post processor for VaadinSecurityAware / VaadinSecurityContextAware

Updated VaadinSecurityConfiguration

### CODE REVIEW

**Comment by peholmst** [comment]
Thanks. I'll have a look at it when I come back from my vacation. In the mean time you could resolve the conflicts that have been introduced after I merged the previous pull requests.

**Comment by GJRTimmer** [comment]
I will look at the conflicts

**Comment by GJRTimmer** [comment]
Added PreAuthorizeViewProviderAccessDelegate to support @PreAuthorize annotation

**Comment by GJRTimmer** [comment]
Conflicts resolved

**Comment by GJRTimmer** [comment]
@peholmst Added spring-vaadin-security/README

Also updated the General project README with a link to the security README.
You can view the final result README from: https://github.com/GJRTimmer/vaadin4spring/tree/security-update/spring-vaadin-security

**Comment by peholmst** [comment]
I was sick the entire last week, so I haven't had time to look a this yet. Will do it this week.

**Comment by GJRTimmer** [comment]
Not a problem.

I will look forward to your comments.

On Mon, Jan 12, 2015 at 12:38 PM, Petter Holmström <notifications@github.com

File: spring-vaadin-security/src/main/java/org/vaadin/spring/security/support/VaadinSecurityAwareProcessor.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package org.vaadin.spring.security.support;
```
Comment (peholmst) [inline spring-vaadin-security/src/main/java/org/vaadin/spring/security/support/VaadinSecurityAwareProcessor.java]: License header, please

File: spring-vaadin-security/src/main/java/org/vaadin/spring/security/provider/PreAuthorizeViewProviderAccessDelegate.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
+package org.vaadin.spring.security.provider;
```
Comment (peholmst) [inline spring-vaadin-security/src/main/java/org/vaadin/spring/security/provider/PreAuthorizeViewProviderAccessDelegate.java]: License header, please

File: spring-vaadin-security/src/main/java/org/vaadin/spring/security/VaadinSecurityContextAware.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package org.vaadin.spring.security;
```
Comment (peholmst) [inline spring-vaadin-security/src/main/java/org/vaadin/spring/security/VaadinSecurityContextAware.java]: License header, please

File: spring-vaadin-security/src/main/java/org/vaadin/spring/security/VaadinSecurityContextAware.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Interface to be implemented by any object that wishes to be notified
+ * of the {@link org.vaadin.spring.security.VaadinSecurityContext}.
+ * 
+ * @author Gert-Jan Timmer (gjr.timmer@gmail.com)
+ *
+ */
+public interface VaadinSecurityContextAware {
```
Comment (peholmst) [inline spring-vaadin-security/src/main/java/org/vaadin/spring/security/VaadinSecurityContextAware.java]: Shouldn't this interface extend the Aware interface?

File: spring-vaadin-security/src/main/java/org/vaadin/spring/security/VaadinSecurityContext.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package org.vaadin.spring.security;
```
Comment (peholmst) [inline spring-vaadin-security/src/main/java/org/vaadin/spring/security/VaadinSecurityContext.java]: License header, please

File: spring-vaadin-security/src/main/java/org/vaadin/spring/security/VaadinSecurityAware.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package org.vaadin.spring.security;
```
Comment (peholmst) [inline spring-vaadin-security/src/main/java/org/vaadin/spring/security/VaadinSecurityAware.java]: License header, please

File: spring-vaadin-security/src/main/java/org/vaadin/spring/security/VaadinSecurity.java [hunk 7]
```diff
@@ -0,0 +1,118 @@
+package org.vaadin.spring.security;
```
Comment (peholmst) [inline spring-vaadin-security/src/main/java/org/vaadin/spring/security/VaadinSecurity.java]: License header, please

**Comment by peholmst** [comment]
Added some minor comments, otherwise looks good! I'll merge it as soon as you've addressed my comments.

**Comment by GJRTimmer** [comment]
@peholmst Processed Comments, Added license Headers, and corrected interface extension.

**Pull request merged**